WordPress/src/main/java/org/wordpress/android/ui/comments/unified/UnifiedCommentListFragment.kt

```diff
@@ -167,7 +173,27 @@ class UnifiedCommentListFragment : Fragment(R.layout.unified_comment_list_fragme
     val commentDetails = registerForActivityResult(CommentDetailsActivityContract()) { response:
     CommentDetailsActivityResponse? ->
         if (response != null) {
-            viewModel.performSingleCommentModeration(response.commentId, response.commentStatus)
+            viewLifecycleOwner.lifecycleScope.launchWhenStarted {
+                if (sharedPreferences?.getString(
+                        COMMENT_STATE_KEY,
+                        COMMENT_STATE_INITIAL
+                    ) == COMMENT_STATE_SUCCESSFULLY_EDITED
+                ) {
+                    viewModel.localCommentCacheUpdateHandler.requestCommentsUpdate()
+
+                    // If the user does a comment moderation next, add a short delay to avoid a
+                    // race condition updating data on the main comment list.
+                    delay(10)
```

Comment:
Delay is a bit of a hacky way to address a race condition, in my opinion. I would try to avoid it, to begin with. Also, looking at the code, it seems that if the comment is moderated and edited at the same time, the `performSingleCommentModeration` should have taken care of updating the list from local cache by calling `requestCommentsUpdate` befiore network request.